ibis/sql/alchemy.py

```diff
@@ -1099,6 +1100,10 @@ def begin(self):
 
     @invalidates_reflection_cache
     def create_table(self, name, expr=None, schema=None, database=None):
+        # reset database if it is the same one used by the connection
+        if database == self.database_name:
```

Comment:
why is this needed?

Reply:
I hit #1979 testing load_data method locally. Just reseting the database variable when it is the same as sef.database_name fixed the issue.

another way to fix that maybe would be replace `self.engine.url.database` by `self.database_name`